MercurialExtensions/fixutf8/win32helper.py

```diff
@@ -233,15 +233,25 @@ def consolehascp():
 	def rawprint(h, s):
 		InternalWriteFile(h, s)
 
-	def getargs():
+	def getUtf8NonConfigArgs():
 		'''
 		getargs() -> [args]
 
 		Returns an array of utf8 encoded arguments passed on the command line.
+		
+		Skips any --config and following arguments since this is used in a method where
+		those arguments are already removed
 		'''
 		c = INT(0)
 		pargv = CommandLineToArgv(GetCommandLine(), byref(c))
-		return [fromunicode(pargv[i]) for i in xrange(1, c.value)]
+		cleanArguments = []
+		iterator = iter(xrange(1, c.value))
+		for i in iterator:
+			if pargv[i] != "--config":
```

Comment:
I don't know python much, but this looks as if it will skip --config, but not any following arguments. There's nothing like a break, it only checks the current one to see whether it IS --config.